### PR TITLE
Blog app: add API endpoints

### PR DESCRIPTION
**Pull request by RolandM99**
### PR requirement:

Create an API endpoint to list all posts for a user.
Create an API endpoint to list all comments for a user's post.
Create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should not be set by passing a parameter.
Add authentication to your API endpoints.
Your API endpoints should receive JSON and respond JSON as well.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -4,6 +4,16 @@
...
+  
+  namespace :api, default: {format: :json} do 
+    namespace :v1,  default: {format: :json} do
+      resources :users, only: [:index, :show] do
+      resources :posts, only: [:index, :show, :create] do
+        resources :comments, only: [:index, :create]
+      end
+    end
+  end
+end
```
Comment (andersonlebon) [inline config/routes.rb]: - [ ] Kindly add the endpoint for `signing up or registering a new user` and the endpoint for `log in`.

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @RolandM99 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [ ] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 1 above)
Comment (RolandM99) [reply config/routes.rb]: Hello @andersonlebon ,
By default authentication is handled by devise, this extends to the API endpoints as well so we do not need to add auth endpoints as devise already handles this for us. But, I tried to expose it. 
Kindly, take into consideration my concern.
 Thank you for your review.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 class ApplicationController < ActionController::Base
-  before_action :authenticate_user!
   protect_from_forgery with: :exception
-
+  before_action :authenticate_with_token
+  # before_action :authenticate_user!
```
Comment (hiromataba) [inline app/controllers/application_controller.rb:4]: - [ ] [OPTIONAL] Please remove commented code to make your code clean

**Review by hiromataba — APPROVED** [review]
# Status: Approved :tada:  :champagne:  :confetti_ball: 
Hi  :wave:   :wave:  @RolandM99 

200w 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
- No Linter errors :heavy_check_mark: 
- Your README file looks very professional :+1: 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**